File: frontend/src/types/match.ts
export interface MatchResultItem {
  id: string;
  job_id: string;
  total_score: number;
  hard_filter_score: number;
  skill_match_score: number;
  semantic_score: number;
  optimization: string;
  is_favorited: boolean;
  created_at: string;
}

export interface MatchResultDetail extends MatchResultItem {
  match_detail: Record<string, unknown>;
  job: {
    title: string;
    company_name: string;
    location_city: string;
    salary_min: number;
    salary_max: number;
    description: string;
    skills_required: string[];
  };
  resume: {
    parsed_data: Record<string, unknown>;
    skills: string[];
  };
}

export interface MatchSession {
  session_id: string;
  status: string;
  total_matches: number;
  results: MatchResultItem[];
}

File: backend/app/schemas/match.py
from datetime import datetime
from pydantic import BaseModel


class MatchRequest(BaseModel):
    resume_id: str


class MatchResultItem(BaseModel):
    id: str
    job_id: str
    total_score: float
    hard_filter_score: float
    skill_match_score: float
    semantic_score: float
    optimization: str
    is_favorited: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class MatchResultDetail(MatchResultItem):
    match_detail: dict
    job: dict
    resume: dict


class MatchSessionResponse(BaseModel):
    session_id: str
    status: str  # processing, completed, failed
    total_matches: int = 0
    results: list[MatchResultItem] = []
